src/main/java/org/openhab/binding/zwave/internal/OrderingSetConverter.java

```diff
@@ -0,0 +1,77 @@
+/**
+ * Copyright (c) 2010-2021 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.zwave.internal;
+
+import java.util.HashSet;
+import java.util.LinkedHashSet;
+import java.util.Set;
+import java.util.stream.Stream;
+
+import org.eclipse.jdt.annotation.Nullable;
+
+import com.thoughtworks.xstream.converters.MarshallingContext;
+import com.thoughtworks.xstream.converters.collections.CollectionConverter;
+import com.thoughtworks.xstream.io.HierarchicalStreamWriter;
+import com.thoughtworks.xstream.mapper.Mapper;
+
+/**
+ * Implements a converter which re-orders set on marshalling
+ *
+ * @author Sami Salonen
+ */
+public class OrderingSetConverter extends CollectionConverter {
+
+    public static boolean allSameClassAndComparable(Set<@Nullable ?> src) {
```

Comment:
Copy-pasted from open openhab-core PR https://github.com/openhab/openhab-core/pull/2437. I think this is OK as both projects have same license and I am contributing the code in both cases.

https://github.com/openhab/openhab-core/blob/f68ca0c8d99b7938d6f7839b9f649cbd3919fb42/bundles/org.openhab.core.config.core/src/main/java/org/openhab/core/config/core/OrderingSetSerializer.java#L35-L50